### PR TITLE
Add slits to WorkflowCreationHelper::create2DWorkspaceWithReflectometryInstrument

### LINKED ISSUE #20965: WorkspaceCreationHelper: add slits for reflectometry instruments
`WorkspaceCreationHelper::create2DWorkspaceWithReflectometryInstrumentMultiDetector` does not account for slits.
Adding two slits to the instrument is generally good for reflectometry instruments.

For example, `NRCalculateSlitResolutionTest` can profit from it. There, `createWorkspace` can be replaced. Test results should not change.

There is some copied code in `create2DWorkspaceWithReflectometryInstrument` and `create2DWorkspaceWithReflectometryInstrumentMultiDetector`.

### Expected behavior

All tests should pass. 

### Platforms affected

All

### PR DESCRIPTION
**Pull request by vreimund**
`WorkspaceCreationHelper::create2DWorkspaceWithReflectometryInstrumentMultiDetector` does not account for slits. Adding two slits to the instrument is generally good for reflectometry instruments.

For example, `NRCalculateSlitResolutionTest` can profit from it. There, `createWorkspace` can be replaced. Test results should not change.

There is some copied code in `create2DWorkspaceWithReflectometryInstrument` and `create2DWorkspaceWithReflectometryInstrumentMultiDetector`.

- All unit tests should pass
- Previous behaviour should not change
- Slits added to reflectometry instruments, will be needed in future gravity correction unit tests

Fixes #20965. 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/TestHelpers/src/WorkspaceCreationHelper.cpp [hunk 1]
```diff
@@ -504,8 +504,16 @@ createEventWorkspaceWithNonUniformInstrument(int numBanks, bool clearEvents) {
...
+void addComponent(Instrument_sptr &instrument, const V3D &position,
+                  std::string name = "component") {
+  ObjComponent *component = new ObjComponent(name);
+  component->setPos(position);
+  instrument->add(component);
+}
+
 void addSource(Instrument_sptr &instrument, const V3D &position,
                std::string name = "source") {
+  addComponent(instrument, position, name);
```
Comment (ianbush) [inline Framework/TestHelpers/src/WorkspaceCreationHelper.cpp]: @reimundILL I think your problem is here - I guess this line was added accidentally!

Comment (reimundILL) [reply Framework/TestHelpers/src/WorkspaceCreationHelper.cpp]: Thank you very much! On my machine that seems to solve the problem, let's see ...

File: Framework/Algorithms/test/NRCalculateSlitResolutionTest.h [hunk 2]
```diff
@@ -4,26 +4,19 @@
...
 
 #include "MantidAlgorithms/NRCalculateSlitResolution.h"
-#include "MantidDataObjects/Workspace2D.h"
-#include "MantidAPI/AnalysisDataService.h"
-#include "MantidAPI/Run.h"
 #include "MantidGeometry/Instrument.h"
-#include "MantidGeometry/Instrument/Detector.h"
-#include "MantidKernel/PropertyWithValue.h"
 #include "MantidKernel/TimeSeriesProperty.h"
-#include "MantidKernel/V3D.h"
```
Comment (ianbush) [inline Framework/Algorithms/test/NRCalculateSlitResolutionTest.h]: With headers the ideal is to 'Use-What-You-Include'. As you use `V3D` in this file it should be left in the header, the same possibly for some others. Removing the truly redundant includes is a nice clean-up though :+1: . See - https://github.com/include-what-you-use/include-what-you-use#instructions-for-users and https://www.mantidproject.org/IWYU .

File: Framework/Algorithms/test/NRCalculateSlitResolutionTest.h [hunk 3]
```diff
@@ -4,26 +4,19 @@
...
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
 
 using namespace Mantid::Algorithms;
 using namespace Mantid::API;
-using namespace Mantid::DataObjects;
-using namespace Mantid::Geometry;
```
Comment (ianbush) [inline Framework/Algorithms/test/NRCalculateSlitResolutionTest.h]: :+1:

File: Framework/Algorithms/test/NRCalculateSlitResolutionTest.h [hunk 4]
```diff
@@ -92,44 +85,13 @@ class NRCalculateSlitResolutionTest : public CxxTest::TestSuite {
...
-    ObjComponent *source = new ObjComponent("source");
-    source->setPos(V3D(0, 0, 0));
-    instrument->add(source);
-    instrument->markAsSource(source);
-
-    Detector *monitor = new Detector("Monitor", 1, nullptr);
-    monitor->setPos(14, 0, 0);
-    instrument->add(monitor);
-    instrument->markAsMonitor(monitor);
+  MatrixWorkspace_sptr createWorkspace(const std::string &name,
```
Comment (ianbush) [inline Framework/Algorithms/test/NRCalculateSlitResolutionTest.h]: :+1:

**Review by ianbush — CHANGES_REQUESTED** [review]
This looks good, a nice tidy up of the code. Can you just double check the headers you have removed, it might be worth giving the IWYU tool a go.

File: Framework/Algorithms/test/NRCalculateSlitResolutionTest.h (hunk 2 above)
Comment (reimundILL) [reply Framework/Algorithms/test/NRCalculateSlitResolutionTest.h]: I will try this tool, thanks for this information!

**Comment by reimundILL** [comment]
A remark (according to initial tests): when creating the two reflectometry workspaces of `WorkspaceCreationHelper` and saving them, then loading into MantidPlot, the instrument is not defined (Show Instrument not highlighted). When selecting Show Detectors, MantidPlot crashes. Maybe the loading or saving is the problem but likely the instrument is not correctly defined. This is not part of this PR but it is interesting since those workspaces are used for testing.

**Comment by reimundILL** [comment]
In `WorkspaceCreationHelper.h` I did not delete #include "MantidAPI/Run.h" on my branch! I see it different here.

**Comment by reimundILL** [comment]
Two conflicting definitions of what is `Left` in ReferenceFrame.h and PeakBoundingBox.h

**Review by ianbush — APPROVED** [review]

**Comment by ianbush** [comment]
This is looking good now, extra changes seem to make sense.

**Pull request merged**